jvm/src/main/scala/com/nawforce/apexlink/cst/VerifyContext.scala

```diff
@@ -388,6 +390,25 @@ abstract class BlockVerifyContext(parentContext: VerifyContext)
 
   def returnType: TypeName
 
+  /** Check if this context or any parent context is within a loop */
+  def isInLoop: Boolean = inLoop || parent()
+    .flatMap {
+      case bvc: BlockVerifyContext => Some(bvc.isInLoop)
+      case _                       => None
+    }
+    .getOrElse(false)
+
+  /** Temporarily set the loop flag for the duration of an operation */
+  def withInLoop[T](op: => T): T = {
+    val previousInLoop = inLoop
+    this.inLoop = true
+    try {
+      op
+    } finally {
+      this.inLoop = previousInLoop
+    }
+  }
```

Comment:
When would it be safe to set the flag on the loop context and leave it ?

My understanding is we don't always create a separate context in the loop verify - therefore we don't want to poison that parent context by setting true loop flag on it. Can we create contexts in the while statements to make a permanent flag work?